### ISSUE TITLE
[BUG] `DD_TAGS` environment variable not considered for log entries in container using `serverless-init`

### ISSUE THREAD

**Issue by mariuskiessling** [DataDog/datadog-agent#15836]
**Agent Environment**
Embedded agent using datadog/serverless-init:beta7 (`sha256:cb24d96298587f7f1fc449107b13181572e96e6439762785afaccff192d1e31a`) resulting in 
_datadog_init_version_ `beta7`.

**Describe what happened:**
When running a container image with the embedded Datadog agent from the `serverless-init` image, the `DD_TAGS` environment variable is not considered when forwarding logs. The environment variable however is considered for e.g. traces and properly translated into span tags.

In `datadog-agent/cmd/serverless-init/main.go`, the init cmd is getting the cloud provider in line 39 and is then using it to derive the tags (line 40). However, looking at `datadog-agent/cmd/serverless-init/cloudrun.go` line 25 where this is implemented, the `DD_TAGS` environment variable seems to be completely ignored. The same is the case for the Azure implementation in `containerapp.go`.

**Describe what you expected:**
The `DD_TAGS` environment variable is parsed and the defined tags are attached to every log line forwarded to Datadog using the embedded agent from the `serverless-init` image.

**Steps to reproduce the issue:**
Build a Docker image with the following files:
* `main.go`:
  ```go
  package main
  
  import (
  	"log"
  	"time"
  )
  
  func main() {
  	for {
  		log.Println("Hello from a serverless init environment.")
  		<-time.After(1 * time.Second)
  	}
  }
  ```
* `Dockerfile`:
  ```Dockerfile
  FROM golang:1.20 as build
  WORKDIR /build
  COPY . .
  RUN go build -o test -ldflags "-s" main.go
  
  FROM gcr.io/distroless/base
  COPY --from=build /build/test /test
  COPY --from=datadog/serverless-init /datadog-init /app/datadog-init
  
  ENV DD_SERVICE=my-service
  ENV DD_ENV=dev
  ENV DD_VERSION=unknown
  ENV DD_TAGS="foo:bar"
  
  ENTRYPOINT ["/app/datadog-init"]
  CMD ["/test"]
  ```

Then run the built container image locally or preferably within Google Cloud Run while making sure to inject a `DD_API_KEY` environment variable with a valid API key.

Navigating to the live log tail of the Datadog organisation the API key was issued for and filtering for 

```
service:my-service
```

will show a new log line every second. Every log entry should have the following tags assigned (adapted for your environment):

* `_dd.origin`: `cloudrun`
* `container_id`: `00f8b6bdb80f86b6fce6c93c1915c130d73048c2b41735b415ca93c9ededdbeef7fb0d7b2dff6e15d7cd6cb016791162daeb6ae3dd023f51ad1f6c5ee68c835a`
* `datadog.index`: `main`
* `datadog_init_version`: `beta7`
* `env`: `dev`
* `location`: `europe-west1`
* `origin`: `cloudrun`
* `project_id`: `my-gcp-project`
* `revision_name`: `my-service-00001-wam`
* `service`: `my-service`
* `service_name`: `my-service`
* `source`: `cloudrun`
* `version`: `unknown`

Note the missing `foo`: `bar` tag/value pair specified in the Dockerfile.

**Additional environment details (Operating System, Cloud provider, etc):**
Google Cloud Run

**Comment by mariuskiessling** [comment]
This issue is connected to support request no. 1112102. I am also happy to contribute a fix if this is considered faulty behaviour.

**Comment by tianchu** [comment]
@IvanTopolcic are you aware of this?

**Comment by mariuskiessling** [comment]
@tianchu / @IvanTopolcic  Do you have any updates on this issue? We would like to continue with our serverless integration of Datadog.

**Comment by hghotra** [comment]
Fixed by #16011
